tools/python/boutiques/tests/test_publisher.py

```diff
@@ -52,6 +62,45 @@ def test_publication(self):
                   "PWr2VeQZgBLErKbfF5RdrKhzzJi8i2hnN8r"])
         self.assertTrue("Desriptor already has a DOI" in str(e.exception))
 
+        # Test publication of an updated version of the same descriptor
+        example1_desc_updated = op.join(example1_dir,
+                                        "example1_docker_updated.json")
+        temp_descriptor_updated = tempfile.NamedTemporaryFile(suffix=".json")
+        shutil.copyfile(example1_desc_updated, temp_descriptor_updated.name)
+
+        with open(temp_descriptor_updated.name, 'r') as fhandle:
+            descriptor_updated = json.load(fhandle)
+
+        # Give updated descriptor the same title as before
+        with open(temp_descriptor_updated.name, 'w') as fhandle:
+            descriptor_updated['name'] = unique_title
+            fhandle.write(json.dumps(descriptor_updated, indent=4,
+                                     sort_keys=True))
+
+        # Publish the updated version
+        new_doi = bosh(["publish",
+                        temp_descriptor_updated.name,
+                        "--sandbox", "-y", "-v",
+                        "--zenodo-token", "hAaW2wSBZMskxpfigTYHcuDrC"
+                        "PWr2VeQZgBLErKbfF5RdrKhzzJi8i2hnN8r"])
+        assert(new_doi)
+
+        # Updated version of descriptor should have a new DOI
+        with open(temp_descriptor_updated.name, 'r') as fhandle:
+            descriptor_updated = json.load(fhandle)
+            assert(descriptor_updated.get('doi') == new_doi)
+            assert(descriptor_updated.get('doi') != doi)
+
+        # Search for the descriptor by its title
+        results = bosh(["search", unique_title, "--sandbox", "-v"])
+
+        # Make sure there is only one result with this title
+        # and that its DOI is the updated one
```

Comment:
So this seems to randomly fail on Travis... sometimes the search doesn't find the descriptor even though in the logs you can see that it was published successfully. Any ideas why?

Reply:
not really, most likely their API is a bit unstable. This is related to #289. @simon-dube told me that good solutions exist to mock servers but I don't remember which ones...

Reply:
Ok, so should I remove the test?

Reply:
Hi @glatard, sorry for the delay. I don't remember using mocks in the CARMIN server project, but a unit test should not depend on remote calls. As a side note, from what I can see, I believe that testing that publishing a descriptor which alrady has a DOI should fail should be moved to its own test.